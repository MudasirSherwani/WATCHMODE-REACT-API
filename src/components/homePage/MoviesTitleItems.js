import React from 'react';
import { useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { RiMovieFill } from 'react-icons/ri';
import { fetchDetailData } from '../../redux/fetch-movies-details';

const MoviesTitleItems = (props) => {
  const { id, title } = props;
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const showDetails = () => {
    dispatch(fetchDetailData(id));
    navigate('/MoviesDetails');
  };

  return (
    <div className="container" data-testid="homepageitem">
      <div className="item">
        <RiMovieFill id={id} size={100} onClick={showDetails} className="movie-icon" alt="movies icon" />
        <h3 className="title">{title}</h3>
      </div>
    </div>
  );
};

MoviesTitleItems.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
export default MoviesTitleItems;
