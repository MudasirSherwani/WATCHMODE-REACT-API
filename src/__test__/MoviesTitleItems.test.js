import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';
import MoviesTitleItems from '../components/homePage/MoviesTitleItems';

it('Check Changes in the Movies Title', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<MoviesTitleItems title="Movie Name Here" id={1} />} />
        </Routes>
      </Router>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check rendering of div', () => {
  render(
    <Provider store={store}>
      <Router>
        <Routes>
          <Route path="/" element={<MoviesTitleItems title="Movie Name Here" id={1} />} />
        </Routes>
      </Router>
    </Provider>,

  );

  const homepage = screen.getByTestId('home-page');

  expect(homepage).toBeInTheDocument();
});
