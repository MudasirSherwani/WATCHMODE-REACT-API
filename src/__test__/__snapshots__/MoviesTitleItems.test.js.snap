// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Changes in the Movies Title 1`] = `
<div
  className="container1"
  data-testid="home-page"
>
  <div
    className="item"
  >
    <svg
      alt="movies icon"
      className="movie-icon"
      fill="currentColor"
      height={50}
      id={1}
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width={50}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        fillRule="evenodd"
      />
    </svg>
    <h3
      className="title-home"
    >
      Movie Name Here
    </h3>
    <h6
      className="year-home"
    >
      Year-
      2022
    </h6>
  </div>
</div>
`;
